RelationalAI/Client.cs

```diff
@@ -147,13 +153,11 @@ public DeleteEngineResponse DeleteEngine(string engine)
 
         public DeleteEngineResponse DeleteEngineWait(string engine)
         {
-            var resp = this.DeleteEngine(engine);
-            var status = resp.Status.State;
-            while (!IsTerminalState(status, "DELETED"))
-            {
-                Thread.Sleep(2000);
-                status = this.GetEngine(resp.Status.Name).State;
-            }
+            var resp = DeleteEngine(engine);
+            resp.Status.State = Policy
+                .HandleResult<Engine>(e => !IsTerminalState(e.State, "DELETED"))
+                .Retry15Min()
```

Comment:
Can we make the `Retry15Min` generic ? For example `Retry(mins)` or `RetryMin(mins)` :pray: